tests/cupy_tests/logic_tests/test_comparison.py

```diff
@@ -97,6 +97,16 @@ def test_array_equal_diff_length(self, xp, dtype):
         b = xp.array([1, 2, 3], dtype=dtype)
         return xp.array_equal(a, b)
 
+    @testing.for_float_dtypes()
```

Comment:
what about complex types?😀

Reply:
Would it make sense to test for all dtypes here too?
https://github.com/cupy/cupy/blob/0dc6bbafb0cf75bb873531cbf69c6995a72c61a3/tests/cupy_tests/logic_tests/test_comparison.py#L145-L147
https://github.com/cupy/cupy/blob/0dc6bbafb0cf75bb873531cbf69c6995a72c61a3/tests/cupy_tests/logic_tests/test_comparison.py#L191-L193

Reply:
If complex arrays are not supported when comparing nans then a `NotImplementedError` should be thrown because NumPy allows it.
In such case the tests should check the complex type and assert that it raises the error.

Reply:
NumPy allows complex arrays with `array_equal`, [ref.](https://github.com/numpy/numpy/blob/ab22e0076608f4825bf9a3ca1b6a0a6a7a670d14/numpy/core/numeric.py#L2431-L2432)
Consecutively, there's a check for complex datatype here:
https://github.com/cupy/cupy/blob/0dc6bbafb0cf75bb873531cbf69c6995a72c61a3/cupy/_logic/comparison.py#L115-L117

But I don't think this case is covered in tests.

Reply:
Tests should cover, but also the `equal_nan` behavior with complex numbers should be explicitly check.